### PR TITLE
Update Release instructions (minor updates)

### PR DESCRIPTION
**Pull request by lakshmimsft**
# Description

Minor updates to current Release Instructions

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

### CODE REVIEW

**Review by ytimocin — APPROVED** [review]

File: docs/contributing/contributing-releases/README.md [hunk 1]
```diff
@@ -103,7 +105,7 @@ Update it to reflect the new release version that we would like to release ([Exa
    git checkout -b <USERNAME>/<BRANCHNAME>
    ```
 
-1. Cherry-pick the `version.json` changes from the previous steps in this PR. This will ensure that the version changes are included in the release branch ([Example](https://github.com/radius-project/bicep/pull/704/files)). You can get the commit hash by running `git log --oneline` in the main branch. PLEASE USE `-x` HERE TO ENSURE VERSION HISTORY IS PRESERVED.
+1. Cherry-pick the `version.json` changes from the previous steps in this PR. This will ensure that the version changes are included in the release branch ([Example](https://github.com/radius-project/bicep/pull/704/files)). You can get the commit hash by running `git log --oneline` in the bicep-extensibility branch. PLEASE USE `-x` HERE TO ENSURE VERSION HISTORY IS PRESERVED.
```
Comment (ytimocin) [inline docs/contributing/contributing-releases/README.md:108]: Is this line supposed to be in the bicep repo?

**Review by ytimocin — DISMISSED** [review]

File: docs/contributing/contributing-releases/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ We follow a monthly release cadence. Any contributions that have been merged thr
 
 ## Release Process
 
+For the entire release process, directly clone repositories under the radius-project organization and create branches off of them. Do not create branches in your personal forks when creating pull requests.
```
Comment (sk593) [inline docs/contributing/contributing-releases/README.md:33]: What's the error that we run into when creating PRs from forks?

**Review by sk593 — DISMISSED** [review]

File: docs/contributing/contributing-releases/README.md (hunk 2 above)
Comment (lakshmimsft) [reply docs/contributing/contributing-releases/README.md:33]: Error was in accessing github token. https://github.com/radius-project/radius/actions/runs/8820651650/job/24214732371

Comment (kachawla) [inline docs/contributing/contributing-releases/README.md:33]: Could you also add a note on why forks won't work.

**Review by kachawla — APPROVED** [review]

File: docs/contributing/contributing-releases/README.md [hunk 3]
```diff
@@ -30,6 +30,8 @@ We follow a monthly release cadence. Any contributions that have been merged thr
 
 ## Release Process
 
+For the entire release process, directly clone repositories under the radius-project organization and create branches off of them. Do not create branches in your personal forks when creating pull requests. This is due to an error accessing the GitHub token when forks are used.
```
Comment (kachawla) [inline docs/contributing/contributing-releases/README.md:33]: @willdavsmith could you remind me if we have an issue tracking this?

**Pull request merged**